qemu/tests/blockdev_stream_cor_base.py

```diff
@@ -36,7 +36,12 @@ def _is_backing_exit(backing, depth):
         image = self.get_image_by_tag(name)
         filename = image.image_filename
         if backing["driver"] == "copy-on-read":
-            actual_filename = backing["file"]["file"]["filename"]
+            cor_opts = backing["file"]["file"]
+            if cor_opts["driver"] == "iscsi":
```

Comment:
def _is_same_file(self, file_params, file_opts):
         # FIXME: this should be supported in VT
         mapping = {'gluster': 'path', 'iscsi': 'lun',
                    'nbd': 'server.port', 'rbd': 'image'}
         option = mapping.get(file_opts["driver"], 'filename')
         return file_params[option] == file_opts[option]

cor_opts = backing['file']['file'] if backing['driver'] == 'copy-on-read' else backing['file']
file_opts = qemu_storage.filename_to_file_opts(filename)
if not self._is_same_file(cor_opts, file_opts):
    self.test.fail(xxx)

We don't want to form the filename in a uri (e.g. iscsi://xxx/xx/0) because there may be auth info in image.image_filename, in that case, the two filename strings are different, but in fact, they are the same.

Reply:
Have updated, thanks zhenchao.